composition-js/src/merging/reporter.ts

```diff
@@ -112,6 +113,7 @@ export class MismatchReporter {
     message: string,
     supergraphElement: TMismatched,
     subgraphElements: (TMismatched | undefined)[],
+    coordinate: string | undefined,
```

Comment:
Same remarks as in `CompositionHint`: we should imo pass `element: NamedSchemaElement | undefined`.

In fact, it's worth double checking, but I suspect we can avoid the need of that argument, at least in most case, as `supergraphElement` is usually what we want here (when it is a `NamedSchemaElement` at least; and when it isn't, we want `undefined`). From a quick look, it seems that there is 2 special cases where that's not quite true, for hints on enum values and union members where `supergraphElement` is the type itself, but we ideally would want to pass the value/member itself. But I'd personally suggest adding an optional argument (say, `targetedElement?: NamedSchemaElement` and only pass it when it's not equal to `supergraphElement` (and default to the later when `targetedElement` is not provided).

Reply:
You were correct, I noticed that while going through and fixing unit tests. I have added the `targetedElement?: NamedSchemaElement` to this function and filled in it for the Enum value cases.